src/client/auth.js

```diff
@@ -1,31 +1,44 @@
-const { MsAuthFlow } = require('./authFlow.js')
+const { Authflow: PrismarineAuth } = require('prismarine-auth')
+const minecraftFolderPath = require('minecraft-folder-path')
+const debug = require('debug')('minecraft-protocol')
 const { uuidFrom } = require('../datatypes/util')
 
 /**
- * Obtains Minecaft profile data using a Minecraft access token and starts the join sequence
+ * Authenticates to Minecraft via device code based Microsoft auth,
+ * then connects to the specified server in Client Options
+ *
+ * @function
  * @param {object} client - The client passed to protocol
  * @param {object} options - Client Options
- * @param {string} chains - Minecraft JWTs to send to server
  */
-async function postAuthenticate (client, options, chains) {
-  // First chain is Mojang stuff, second is Xbox profile data used by mc
-  const jwt = chains[1]
-  const [header, payload, signature] = jwt.split('.').map(k => Buffer.from(k, 'base64')) // eslint-disable-line
-  const xboxProfile = JSON.parse(String(payload))
-
-  // This profile / session here could be simplified down to where it just passes the uuid of the player to encrypt.js
-  // That way you could remove some lines of code. It accesses client.session.selectedProfile.id so /shrug.
-  // - Kashalls
-  const profile = {
-    name: xboxProfile?.extraData?.displayName || 'Player',
-    uuid: xboxProfile?.extraData?.identity || 'adfcf5ca-206c-404a-aec4-f59fff264c9b', // random
-    xuid: xboxProfile?.extraData?.XUID || 0
+async function authenticate (client, options) {
+  if (!options.profilesFolder) {
+    options.profilesFolder = minecraftFolderPath
```

Comment:
Should be minecraftFolderPath + '/nmp-cache'

Reply:
I should probably implement this fix into nmp as well.